src/location/util.rs

```diff
@@ -15,51 +15,504 @@
+                Err(e) => {
+                    table_entry.partition_info = None;
+                    error!(
+                        "Location::get_table_entry_from_remote_inner fetch_partition_info error:{}",
+                        e
+                    );
+                    return Err(CommonErr(
+                        CommonErrCode::PartitionError,
+                        format!(
+                            "location::get_table_entry_from_remote_inner fetch_partition_info error:{e}",
+                        ),
+                    ));
+                }
+            }
+            if let Some(v) = table_entry.partition_info.clone() {
+                // fetch first part
+                if let Some(first_part_desc) = &v.first_part_desc {
+                    let ob_part_func_type = first_part_desc.get_part_func_type();
+                    LocationUtil::fetch_first_part(conn, &mut table_entry, ob_part_func_type)?;
+                }
+                // fetch sub part
+                if let Some(sub_part_desc) = &v.sub_part_desc {
+                    let ob_part_func_type = sub_part_desc.get_part_func_type();
+                    LocationUtil::fetch_sub_part(conn, &mut table_entry, ob_part_func_type)?;
+                }
+            }
+            if let Some(v) = &mut table_entry.partition_info {
+                v.part_name_id_map = LocationUtil::build_part_name_id_map(v);
+            }
+        }
+
+        let part_entry = LocationUtil::get_table_location_from_remote(conn, key, &table_entry)?;
+        table_entry.partition_entry = Some(part_entry);
+        table_entry.set_refresh_time_mills(Utc::now().timestamp_millis());
+
+        Ok(table_entry)
+    }
+
+    pub fn get_table_entry_from_conn(
+        conn: &mut my::PooledConn,
+        sql: String,
+        key: &TableEntryKey,
+    ) -> Result<TableEntry> {
+        let mut table_id = OB_INVALID_ID;
+        let mut replica_num = OB_INVALID_ID;
+        let mut partition_num = OB_INVALID_ID;
+        let mut replica_locations = Vec::new();
+
+        for row in conn.query::<Row, String>(sql.clone())? {
+            let (
+                partition_id,
+                svr_ip,
+                sql_port,
+                tbl_id,
+                role,
+                repl_num,
+                part_num,
+                svr_port,
+                status,
+                stop_time,
+                replica_type,
+            ) = match my::from_row_opt(row) {
+                Ok(tuple) => tuple,
+                Err(e) => {
+                    error!("ObTableLocation::get_table_entry_from_conn: fail to do mysql row conversion, err:{}", e);
+                    return Err(CommonErr(
+                        CommonErrCode::ConvertFailed,
+                        format!("mysql row conversion err:{e}"),
+                    ));
+                }
+            };
+            // unused
+            let _: i64 = partition_id;
```

Comment:
Remove it and change L417 to `partition_id: _` if not used?

Reply:
We need to claim the type of partition_id in that position. If we do not assign here(Compiler will do the type inference), we need to write a long list to claim the type of all the results, which is not elegant.